contracts/pool-templates/base/SwapTemplateBase.vy

```diff
@@ -868,6 +861,7 @@ def withdraw_admin_fees():
 
 
 @external
+@nonreentrant('lock')
```

Comment:
In general I feel re-entrancy locks within our non-ETH pools are overkill, as we personally verify tokens prior to pool creation. That said, for the comfort of users it's good to keep them on public methods for swapping and adding/removing liquidity.   But for guarded admin functions, I don't think they're necessary.